docs/analyzer/checker_and_analyzer_configuration.md

```diff
@@ -64,6 +64,19 @@ or source code for more details about the configuration options.
 | graph-trim-interval                   | 1000              |                                          |                                                                                                     |
 
 
+## Z3 Theorem Prover
+The static analyzer supports using the
+[Z3 Theorem Prover](https://github.com/Z3Prover/z3) from Microsoft Research as
+an external constraint solver. This allows reasoning over more complex queries,
+but performance is `~15x` slower than the default range-based constraint
+solver. To enable the Z3 solver backend, Clang must be built with the
+`CLANG_ANALYZER_BUILD_Z3=ON` option, and the
+`-Xanalyzer -analyzer-constraints=z3` arguments passed at runtime.
```

Comment:
Please extend the documentation a bit more that the the user does not have to forward these arguments manually, but CodeChecker automatically detects if the used clang is built with z3 support.